src/variant.rs

```diff
@@ -378,10 +475,164 @@ impl<T: ToVariant> From<T> for Variant {
+                    None => return None,
+                }
+            }
+        }
+
+        Some(vec)
+    }
+}
+
+impl<T: StaticVariantType + ToVariant> ToVariant for Vec<T> {
+    fn to_variant(&self) -> Variant {
+        let mut vec = vec![];
+        for child in self {
+            vec.push(child.to_variant());
+        }
+        Variant::new_array::<T>(&vec)
+    }
+}
+
+impl<T: StaticVariantType> StaticVariantType for Vec<T> {
+    fn static_variant_type() -> Cow<'static, VariantTy> {
+        <[T]>::static_variant_type()
+    }
+}
+
+impl<K, V, H> FromVariant for HashMap<K, V, H>
+where
+    K: FromVariant + Eq + Hash,
+    V: FromVariant,
+    H: BuildHasher + Default,
+{
+    fn from_variant(variant: &Variant) -> Option<Self> {
+        let mut map = HashMap::default();
+
+        let n_children = variant.n_children();
+        if n_children != 0 {
+            for i in 0..n_children {
+                let entry = variant.get_child_value(i);
+                let key = match entry.get_child_value(0).get() {
+                    Some(key) => key,
+                    None => return None,
+                };
+                let val = match entry.get_child_value(1).get() {
+                    Some(val) => val,
+                    None => return None,
+                };
+
+                map.insert(key, val);
+            }
+        }
+
+        Some(map)
+    }
+}
+
+impl<K, V> ToVariant for HashMap<K, V>
+where
+    K: StaticVariantType + ToVariant + Eq + Hash,
+    V: StaticVariantType + ToVariant,
+{
+    fn to_variant(&self) -> Variant {
+        let mut vec = vec![];
+        for (key, value) in self {
+            let entry = Variant::new_dict_entry(
+                &key.to_variant(),
+                &value.to_variant(),
+            );
+            vec.push(entry);
+        }
+        Variant::new_array::<DictEntry<K, V>>(&vec)
+    }
+}
+
+struct DictEntry<K, V>(K, V);
+
+impl<K: StaticVariantType, V: StaticVariantType> StaticVariantType for DictEntry<K, V> {
+    fn static_variant_type() -> Cow<'static, VariantTy> {
+        let key_type = K::static_variant_type();
+        let value_type = V::static_variant_type();
+        let signature = format!("{{{}{}}}", key_type.to_str(), value_type.to_str());
+
+        VariantType::new(&signature).expect("incorrect signature").into()
+    }
+}
+
+impl<K: StaticVariantType, V: StaticVariantType, H: BuildHasher + Default> StaticVariantType for HashMap<K, V, H> {
+    fn static_variant_type() -> Cow<'static, VariantTy> {
+        let key_type = K::static_variant_type();
+        let value_type = V::static_variant_type();
+        let signature = format!("a{{{}{}}}", key_type.to_str(), value_type.to_str());
+
+        VariantType::new(&signature).expect("incorrect signature").into()
+    }
+}
+
+macro_rules! tuple_impls {
+    ($($len:expr => ($($n:tt $name:ident)+))+) => {
+        $(
+            impl<$($name),+> StaticVariantType for ($($name,)+)
```

Comment:
The from/to traits are missing here still

Reply:
I know. I didn't say I'm done. :)